### PR TITLE
[IMP] mail: discuss sidebar quick search

### PR DESCRIPTION
**Pull request by alexkuhn**
Screenshot 2022-12-14 at 13 24 36

Also unskip a single unrelated test

### CODE REVIEW

File: addons/mail/static/src/new/discuss/sidebar.js [hunk 1]
```diff
@@ -115,4 +116,14 @@ export class Sidebar extends Component {
...
         });
     }
+
+    filteredThreads(category) {
+        if (!this.state.quickSearchVal) {
+            return category.threads;
+        }
+        return category.threads.filter((threadLocalId) => {
+            const thread = this.messaging.state.threads[threadLocalId];
+            return thread?.name.includes(this.state.quickSearchVal);
```
Comment (seb-odoo) [inline addons/mail/static/src/new/discuss/sidebar.js]: thread can be undefined here?

Comment (seb-odoo) [reply addons/mail/static/src/new/discuss/sidebar.js]: It looks like it should not be possible if state and relations are properly managed

Comment (seb-odoo) [reply addons/mail/static/src/new/discuss/sidebar.js]: `thread.name` however can be empty for sure, when using group chat and name comes from name of members (should use `displayName`).

File: addons/mail/static/src/new/discuss/sidebar.xml [hunk 2]
```diff
@@ -21,6 +21,7 @@
             </t>
         </div>
         <hr class="w-100 border-top"/>
+        <input t-if="Object.values(messaging.state.threads).filter(thread => thread.is_pinned and thread.model === 'mail.channel').length > 19" class="form-control mx-4 mb-2 rounded-3 w-auto" placeholder="Quick search..." t-model="state.quickSearchVal"/>
```
Comment (seb-odoo) [inline addons/mail/static/src/new/discuss/sidebar.xml]: condition is a bit too complex for my taste to be on a template

File: addons/mail/static/src/new/discuss/sidebar.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/sidebar.js]: name may not exist

Comment (seb-odoo) [reply addons/mail/static/src/new/discuss/sidebar.js]: but then the `?` is misplaced

**Review by seb-odoo — APPROVED** [review]
Ok except above comments

File: addons/mail/static/src/new/discuss/sidebar.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/sidebar.js]: In the end just removed it. Threads always have a name (old tests are being adapted, thanks to props validation of autoresize input)

File: addons/mail/static/src/new/discuss/sidebar.xml (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/discuss/sidebar.xml]: introduced getter `hasQuickSearch`

**Comment by alexkuhn** [comment]
Changes done

**Pull request merged**